packages/data/lib/di/injector_data.dart

```diff
@@ -16,33 +21,64 @@ void initInjectorData() {
 }
 
 void _initModuleApi() {
-  GetIt.instance.registerSingleton<Dio>(
-    _buildMovieDio(
-      GetIt.instance.get<HeaderInterceptor>(),
+  GetIt.instance.registerFactory<Dio>(
+    () => _buildMovieDioTraktApi(
+      GetIt.instance.get<HeaderInterceptorTraktApi>(),
     ),
+    instanceName: traktApi,
   );
-  GetIt.instance.registerSingleton<ApiBaseService<ServicePayLoad>>(
-    ApiBaseServiceImpl(
-      GetIt.instance.get<Dio>(),
+  GetIt.instance.registerFactory<ApiBaseService<ServicePayLoad>>(
+    () => ApiBaseServiceImpl(
+      GetIt.instance.get(
+        instanceName: traktApi,
```

Comment:
not necessarily, but if you create only one instance, then it is better to use generic

Reply:
registration of two identical, but a distinguishing feature they have a name